docs/csharp/language-reference/operators/equality-operators.md

```diff
@@ -0,0 +1,79 @@
+---
+title: "Equality operators - C# Reference"
+ms.date: 03/28/2019
+author: pkulikov
+f1_keywords: 
+  - "==_CSharpKeyword"
+  - "!=_CSharpKeyword"
+helpviewer_keywords: 
+  - "equality operator [C#]"
+  - "equals operator [C#]"
+  - "== operator [C#]"
+  - "inequality operator [C#]"
+  - "not equals operator [C#]"
+  - "!= operator [C#]"
+---
+# Equality operators (C# Reference)
+
+The `==` (equality) and `!=` (inequality) operators check if their operands are equal or not.
+
+## Equality operator ==
+
+The equality operator `==` returns `true` if its operands are equal, `false` otherwise.
+
+### Value types equality
+
+Operands of the [built-in value types](../keywords/value-types-table.md) are equal if their values are equal:
+
+[!code-csharp-interactive[value types equality](~/samples/snippets/csharp/language-reference/operators/EqualityAndNonEqualityExamples.cs#ValueTypesEquality)]
+
+> [!NOTE]
+> For equality and relational operators `==`, `>`, `<`, `>=`, and `<=`, if any of the operands is not a number (<xref:System.Double.NaN?displayProperty=nameWithType> or <xref:System.Single.NaN?displayProperty=nameWithType>), the result of operation is `false`. That means that the `NaN` value is neither greater than, less than, nor equal to any other `double` (or `float`) value. For more information and examples, see the <xref:System.Double.NaN?displayProperty=nameWithType> or <xref:System.Single.NaN?displayProperty=nameWithType> reference article.
```

Comment:
Not related to you PR. We should add to this note that `double.NaN` is not equal to `double.NaN`: